src/controller/AbstractDnssdDiscoveryController.cpp

```diff
@@ -25,6 +25,32 @@
 namespace chip {
 namespace Controller {
 
+bool compareIpAddresses(const size_t sourceNumIPs, const size_t destinationNumIPs, const Inet::IPAddress * source,
```

Comment:
Why is this a global function not scoped to this file?

Reply:
I didn't think I would ever use it anywhere else.
The purpose of this function is to compare the IPAddress List in this file, and the code length would be too long to implement in OnNodeDiscovered.

Reply:
Ah.. I will add "static" keyword in this function.